### ISSUE TITLE
Always failure for  status: '429',     response: { reason: 'TooManyProviderTokenUpdates'}

### ISSUE THREAD

**Issue by li8911255** [node-apn/node-apn#590]
Everything was fine on localhost,but when I run the same code on my remote server,whatever the devicetoken is, the apns result always failed and show me  status: '429',     response: { reason: 'TooManyProviderTokenUpdates'}

**Comment by andrew804** [comment]
@li8911255 have you found what was causing this? I'm not using node-apn but I've got the same problem. Only happens in development environment not in production.

**Comment by mjellis** [comment]
See best practices for managing connections. 
https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingwithAPNs.html#//apple_ref/doc/uid/TP40008194-CH11-SW1

**Comment by caipirginka** [comment]
I'm not entirely sure of this, but I had the same problem on my remote server, while locally it worked.
Then I realized the clock of the server was one hour ahead, don'tknow why...
I set the clock back to correct time, and now it seems the "TooManyProviderTokenUpdates" error went away.
Since the JWT token issued by the APNs server has a one hour expiration validity, it may be that node-apn always found it expired (since the clock was off by one hour) and regenerate it each time.
